automation/WinFormsAutomationApp/AuthenticationHelper.cs

```diff
@@ -191,6 +243,16 @@ public static string ToJson(this object obj)
         #endregion
 
         #region Private Methods
+        // This function clears cookies from the browser control used by ADAL.
+        public static void ClearCookies()
+        {
+            const int INTERNET_OPTION_END_BROWSER_SESSION = 42;
+            InternetSetOption(IntPtr.Zero, INTERNET_OPTION_END_BROWSER_SESSION, IntPtr.Zero, 0);
+        }
+
+        [DllImport("wininet.dll", SetLastError = true)]
+        private static extern bool InternetSetOption(IntPtr hInternet, int dwOption, IntPtr lpBuffer, int lpdwBufferLength);
```

Comment:
Consider moving and consolidating all the P/Invoke code into a 'Native' directory/file or similar.